plugin/src/automation/lift/liftrules.ml

```diff
@@ -159,35 +163,78 @@ let is_packed_constr c env sigma trm =
          let constr = constrs.(i - 1) in
          let constr_f = first_fun (unpack (zoom_term zoom_lambda_term env constr)) in
          if equal constr_f f && List.length args = arity constr then
-           sigma, Some (i - 1, args)
+           sigma, Some (i - 1, args, trm)
```

Comment:
Remove the eta logic here, since it was a temporary attempt at something we don't need anymore